src/java.base/share/classes/java/lang/reflect/Executable.java

```diff
@@ -280,18 +281,20 @@ public Set<AccessFlag> accessFlags() {
      * underlying executable takes no parameters.  Note that the
      * constructors of some inner classes may have an implicitly
      * declared parameter in addition to explicitly declared ones.
-     * Also note that as a <a
-     * href="{@docRoot}/java.base/java/lang/reflect/package-summary.html#LanguageJvmModel">modeling
+     * Also note that compact constructors of a record class may have
+     * implicitly declared parameters.
+     * Finally note that as a <a href="{@docRoot}/java.base/java/lang/reflect/package-summary.html#LanguageJvmModel">modeling
      * artifact</a>, the number of returned parameters can differ
      * depending on whether or not generic information is present. If
-     * generic information is present, only parameters explicitly
-     * present in the source will be returned; if generic information
-     * is not present, implicit and synthetic parameters may be
-     * returned as well.
+     * generic information is present, only parameters implicitly or
```

Comment:
It may be helpful to link to javax.lang.model.util.Elements.Origin.MANDATED and  javax.lang.model.util.Elements.Origin.SYNTHETIC to give the reader some more context here. If you do that, you may need to add some SuppressWarnings to make the compile-time doclint checking work due to the cross-module linkage.

Reply:
> It may be helpful to link to javax.lang.model.util.Elements.Origin.MANDATED and javax.lang.model.util.Elements.Origin.SYNTHETIC to give the reader some more context here. If you do that, you may need to add some SuppressWarnings to make the compile-time doclint checking work due to the cross-module linkage.

PS Under a separate bug, we could add "compact constructor parameters" to the list of mandated constructs in Origin.MANDATED.